### ISSUE TITLE
Local running of libraries tests is broken

### ISSUE THREAD

**Issue by janvorli** [dotnet/runtime#95762]
The recent PR #94868 has broken local running of libraries tests. After each test suite is completed, the following is printed to console and the test suite is reported as failed:
```
XUnitLogChecker.dll does not exist in the expected location: \XUnitLogChecker.dll
```

Looking at the [eng/testing/RunnerTemplate.cmd](https://github.com/dotnet/runtime/blob/283ee4e354c8897f1ad2bf351e45fbfec41ca3df/eng/testing/RunnerTemplate.cmd#L117), I can see that it builds the XUnitLogChecker.dll path like this:
`set XUNITLOGCHECKER_DLL=%HELIX_CORRELATION_PAYLOAD%\XUnitLogChecker.dll`
My guess is the problem is that the HELIX_CORRELATION_PAYLOAD is not set for local testing.

**Comment by carlossanlop** [comment]
Thanks for reporting. Can you please tell me the command you were using to run the tests?

**Comment by janvorli** [comment]
As a repro, you can use e.g. these steps locally. The last step runs the tests and you'll see the suites reported as failing:
* `.\build.cmd clr+libs -c Release -rc Checked`
* `.\build.cmd libs.tests -c Release -rc Checked` 
* `.\build.cmd libs.tests -c Release -rc Checked -test -testnobuild`

**Comment by BruceForstall** [comment]
@carlossanlop I ran into this today (also in local testing).

Do you really need to have a variable `__IsXUnitLogCheckerSupported`? You could presumably skip all this stuff if `HELIX_CORRELATION_PAYLOAD` is not defined, as Jan suggested above.

Also, why does the test script `RunTests.cmd` need `set __TestArchitecture=x86`? Who uses this information? The `RunTests.cmd` script should be independent of processor architecture. The problem now is that I use the same RunTests.cmd script for both x86 and x64 test runs (they take a `-r` parameter to the Core_Root that will be used, and I can pass either an x86 or x64 Core_Root. But now there is an architecture hard-coded into the RunTests.cmd script).

**Comment by ericstj** [comment]
@carlossanlop I just hit this today as well.   My workflow was:
```
git pull upstream main
rd /q /s artifacts
build -rc Release clr+libs
cd src\libraries\System.Runtime\tests\System.Reflection.Tests
dotnet build /t:Test
```

**Comment by carlossanlop** [comment]
Looking at this today.

**Comment by carlossanlop** [comment]
> _Also, why does the test script `RunTests.cmd` need `set __TestArchitecture=x86`? Who uses this information?_

@BruceForstall The debuggers fail to run if the environment variable is not set. It is expected here:

https://github.com/dotnet/runtime/blob/499e2885ba3ff608950c38fa25ba1c1fc5169092/src/tests/Common/Coreclr.TestWrapper/CoreclrTestWrapperLib.cs#L243

> _The problem now is that I use the same RunTests.cmd script for both x86 and x64 test runs (they take a -r parameter to the Core_Root that will be used, and I can pass either an x86 or x64 Core_Root. But now there is an architecture hard-coded into the RunTests.cmd script)._

Well, the value is not hardcoded. It is set in these 3 places:

- CoreCLR tests: 
  - https://github.com/dotnet/runtime/blob/499e2885ba3ff608950c38fa25ba1c1fc5169092/src/tests/Common/helixpublishwitharcade.proj#L712
  - https://github.com/dotnet/runtime/blob/499e2885ba3ff608950c38fa25ba1c1fc5169092/src/tests/Common/helixpublishwitharcade.proj#L712

- Libraries tests:
  - https://github.com/dotnet/runtime/blob/499e2885ba3ff608950c38fa25ba1c1fc5169092/eng/testing/tests.targets#L78

But with [the fix I just submitted](https://github.com/dotnet/runtime/pull/96638), these env vars should not be needed at all in local runs.

**Comment by BruceForstall** [comment]
`__TestArchitecture` should not be set in the generated `RunTests.cmd`. If I build x86, then build x64, the generated RunTests.cmd will be overwritten with a new test architecture setting. This is incorrect.

Note that in the CoreCLR tests, the `__TestArchitecture` variable is only set on Helix. It is not set in any generated wrapper file used by developers on local machines. Libraries testing should do the same.

Similarly, `set __IsXUnitLogCheckerSupported=1` should not be added to the generated RunTests.cmd, even if you do add code to skip it if the HELIX variable is not set.

Note that `__TestArchitecture` is only used to choose a version of cdb.exe to use for dump file processing. This isn't going to work on local machines anyway, since it depends on hard-coded cdb.exe install path (and I, for instance, don't have one at that location). It seems like a hack to set an environment variable to figure this out. Couldn't the wrapper script detect the architecture run type? @jkoritzinsky ?

**Comment by carlossanlop** [comment]
I merged the PR to unblock folks testing locally. Reopening as there is still some discussion to address related to the OS.

**Comment by BruceForstall** [comment]
@carlossanlop I'm suggesting something like this:

https://github.com/dotnet/runtime/pull/96713

(The `IsXUnitLogCheckerSupported` variable might need to be passed through more layers, and maybe there are additional conditions that need to be checked)

**Comment by carlossanlop** [comment]
Thanks for submitting a PR. I suppose the main problem reported by this issue has been fixed, so we can close it. We can just track the additional change in the new PR directly.
